examples/hello-world/gnn-pt/jobs/gnn-pt/app/custom/graphsagetrainer.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#### change to GNN
+import os.path
+import os.path as osp
+
+import torch
+from pt_constants import PTConstants
+from torch import nn
+from torch.optim import SGD
+from nvflare.apis.dxo import DXO, DataKind, MetaKey, from_shareable
+from nvflare.apis.executor import Executor
+from nvflare.apis.fl_constant import ReservedKey, ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable, make_reply
+from nvflare.apis.signal import Signal
+from nvflare.app_common.abstract.model import make_model_learnable, model_learnable_to_dxo
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_opt.pt.model_persistence_format_manager import PTModelPersistenceFormatManager
+import torch.nn.functional as F
+from torch_geometric.loader import DataLoader,LinkNeighborLoader
+from torch_geometric.datasets import PPI
+from torch_geometric.data import Batch
+from torch_geometric.nn import GraphSAGE
+import tqdm
```

Comment:
make sure `tqdm` is in "requirements.txt" as well

Reply:
added